scout/server/blueprints/institutes/templates/overview/phenomodel.html

```diff
@@ -0,0 +1,363 @@
+    </div>
+    <div class="form-group col-2">
+      <button type="submit" name="add_hpo" class="btn btn-secondary btn-sm">Add HPO term</button>
+    </div>
+  </div>
+{% endmacro %}
+
+{% macro term_title(term_type) %}
+  <div class="row d-flex align-items-center">
+    <div class="form-group col-4">
+    {% if phenomodel.subpanels %}
+      <select name="{{term_type}}_subpanel_id" class="form-control" required>
+        <option value="" disabled selected>Select panel</option>
+        {% for subpanel_id, subpanel in phenomodel.subpanels.items() %}
+        <option value="{{subpanel_id}}">{{subpanel.title}}</option>
+        {% endfor %}
+      <select>
+    {% endif %}
+    </div>
+    <div class="form-group col-2">
+      <input type="checkbox" id="{{term_type}}HasTitle" name="{{term_type}}HasTitle">
+      <label class="form-check-label" for="{{term_type}}HasTitle">term has a title</label>
+    </div>
+    <div class="form-group col-5">
+      <input type="text" class="form-control" name="{{term_type}}TermTitle" id="{{term_type}}TermTitle" placeholder="Term title" disabled>
+    </div>
+  </div>
+{% endmacro %}
+
+{% macro subpanels_macro()%}
+  {%if phenomodel.subpanels %}
+    {% for subpanel_id, subpanel in phenomodel.subpanels.items()%}
+      <div class="card">
+        <div class="card-header">
+          <strong>{{subpanel.title}}</strong>&nbsp;&nbsp;<text class="text-muted">{{subpanel.subtitle}}</text>
+        </div>
+        <div class="card-body">
+
+          <div class="row">
+            {% if "checkboxes" in subpanel %}
+              <table class="table table-striped">
+                {% if "checkboxes" in subpanel %}
+                  Checkbox groups:
+                  {% for term_id, term_obj in subpanel.checkboxes.items() %}
+                    <tr>
+                      <td>
+                        <div>
+                          <form action="{{ url_for('overview.checkbox_edit', institute_id=institute._id, model_id=phenomodel._id)}}" method="POST">
+                            {% if term_obj.children %}
+                              <a href="#" onclick="toggleDiv('{{subpanel_id}}#{{term_id}}')">{{term_obj.custom_name or term_obj.description}}</a> ({{term_id}})
+                            {% else %}
+                              {{term_obj.custom_name or term_obj.description}} ({{term_id}})
+                            {% endif %}
+                            <!--button to remove a checkbox group-->
+                            <button type="submit" class="button btn btn-danger btn-xs"  style="float: right;" name="checkgroup_remove" value="{{term_id}}#{{subpanel_id}}"><i class="fa fa-times"></i></button>
+                          </form>
+                        </div>
+                        <div style="margin-left: 35px; display: none;" id="{{subpanel_id}}#{{term_id}}">
+                          {% for item in term_obj.children recursive %}
```

Comment:
Recursive! 😎

Reply:
The only way to loop over a dictionary of undefined nested levels, I found out!